### PR TITLE
Improve a TAIT error and add an error code plus documentation

### PR DESCRIPTION
**Pull request by oli-obk**
cc https://github.com/rust-lang/rust/issues/106858

### CODE REVIEW

File: compiler/rustc_error_codes/src/error_codes/E0792.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+Type alias impl trait can only have their hidden type set
```
Comment (compiler-errors) [inline compiler/rustc_error_codes/src/error_codes/E0792.md:1]:
```suggestion
A type alias impl trait can only have its hidden type defined
```

**Review by compiler-errors — APPROVED** [review]
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @compiler-errors (or someone else) soon.


r=me with nits applied if u want

File: compiler/rustc_error_codes/src/error_codes/E0792.md (hunk 1 above)
Comment (compiler-errors) [reply compiler/rustc_error_codes/src/error_codes/E0792.md:1]: or maybe "assigned"?

Comment (oli-obk) [reply compiler/rustc_error_codes/src/error_codes/E0792.md:1]: I'll go with `assigned`. Opinions at https://github.com/rust-lang/reference/pull/1317/files#diff-5aaceb2a1fd048b1730a7dcaff82d85a05fd5c64faa4ad9df1773adfcd6cf2c1R75 are very welcome ^^

File: tests/ui/consts/const_in_pattern/issue-73431.stderr [hunk 2]
```diff
@@ -1 +1 @@
-WARN rustc_mir_build::thir::pattern::const_to_pat MIR const-checker found novel structural match violation. See #73448.
+2:rustcWARN rustc_mir_build::thir::pattern::const_to_pat MIR const-checker found novel structural match violation. See #73448.
```
[the remark this answers is not in the document]
Comment (oli-obk) [reply tests/ui/consts/const_in_pattern/issue-73431.stderr:1]: grml... parallel compiler

**Pull request merged**